src/main/java/cinemas/repositories/impl/UserRepositoryImpl.java

```diff
@@ -2,11 +2,24 @@
 
 import cinemas.models.User;
 import cinemas.repositories.UserRepository;
+import jakarta.persistence.Query;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository("userRepository")
 public class UserRepositoryImpl extends BaseRepositoryImpl<User, Integer> implements UserRepository {
     public UserRepositoryImpl() {
         super(User.class);
     }
+
+    @Override
+    public User findByEmail(String email) {
+        String hql = "FROM User u WHERE u.email = :email";
+        Query query = entityManager.createQuery(hql);
+        query.setParameter("email", email);
+        query.setMaxResults(1);
+        List<User> users = query.getResultList();
+        return users.isEmpty() ? null : users.get(0);
```

Comment:
Vừa maxx 1 record, rồi lại getListResult  xong lại get 0
Khá nhọc nhằn, sao em ko lấy first record nhỉ

Reply:
Tại em không muốn catch exception chị ạ

Reply:
có thể dùng try ~ catch

Reply:
Nhưng em dùng getResultList là ko hợp lý vì nếu tính Perfomance nhé

Reply:
đâu cần phải try cactch nhỉ, , vẫn dùng toán tử 3 ngôi dc nhá

Reply:
khi em đảm bảo răng email là unit thì rõ ràng em nên dùng uniqueResult() là hặp lý nhất, em cacth ở service là được nhá 

```
Trả về: Nó trả về một đối tượng duy nhất từ kết quả của truy vấn.
Kết quả:
Nếu truy vấn không trả về kết quả nào, nó sẽ trả về null.
Nếu có nhiều hơn một kết quả, nó sẽ ném ra ngoại lệ NonUniqueResultException.
```